### PR TITLE
Fix incorrect regex pattern in `KernelDB`

### PR DESCRIPTION
**Pull request by rdspring1**
Fixes Issue #507

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

File: csrc/kernel_db/kernel_db.cpp [hunk 1]
```diff
@@ -118,8 +118,16 @@ bool KernelDb::open(
...
+        //  --- Group 1: any word character and dash
+        // compile_args
+        //  --- Group 2: any word character, space, plus, dash and equals
+        // kernel_code_file
+        //  --- Group 3: [any word character, dash and slash].cu
+        // cubin_file
+        //  --- Group 4: [any word character, dash and slash].cubin
         std::regex db_line_regex(
-            R"(^([\w-]+),([\w -\=]+),([\w-\/]+\.cu),([\w-\/]+\.cubin)$)");
+            R"(^([\w-]+),([\w \+\-\=]+),([\w\-\/]+\.cu),([\w\-\/]+\.cubin)$)");
```
[the remark this answers is not in the document]
Comment (rdspring1) [reply csrc/kernel_db/kernel_db.cpp:130]: The regex string is parsing this string `kernel_signature,compile_args,kernel_code_file,cubin_file`.
The previous syntax ` -\=` specified the range between space and `=`.  For the command line parameters, we only need space, plus, dash, and equals. Defining four characters explicitly seemed clearer than the range.

[the remark this answers is not in the document]
Comment (jjsjann123) [reply csrc/kernel_db/kernel_db.cpp:130]: Ha, thanks. I didn't realize it's switching from range to individual characters.

**Comment by zasdfgbnm** [comment]
Just curious: why does this depend on gcc version? Is it some compiler bug? (If you don't already know why, there is no need to look into it)

**Pull request merged**